lib/rules/no-array-prototype-extensions.js

```diff
@@ -717,9 +722,10 @@ module.exports = {
         // ignore ember array initialization with A([]) or EmberArray([])
         if (
           node.type === 'CallExpression' &&
-          nodeInitializedTo &&
           nodeInitializedTo.callee &&
-          (nodeInitializedTo.callee.name === 'A' || nodeInitializedTo.callee.name === 'EmberArray')
+          importedEmberArrayName &&
+          nodeInitializedTo.callee.name &&
```

Comment:
Better to check type so we know what type of node we're dealing with instead of just the presence of properties:

```suggestion
          nodeInitializedTo.callee.type === 'Identifier' &&
```

And an Identifier will always have a name.